---
title: "Class 13: Genome Informatics (Part 1)"
output: github_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

##Sample genotypes in the MXL 1000 Genomes Project data

Here we focuse on the Mexican Ancestry in Los Angeles, California (MXL) population.

What proportion of the Mexican Ancestry in Los Angeles sample population (MXL) are homozygous for the asthma associated SNP (G|G)? 

```{r}
mxl <- read.csv("373531-SampleGenotypes-Homo_sapiens_Variation_Sample_rs8067378.csv")
head(mxl)
```

How many of each genotype are there?
```{r}
table(mxl$Genotype..forward.strand.)
```

Proportion or percent of total for each genotype?
```{r}
(table(mxl$Genotype..forward.strand.) / nrow(mxl)) *100
```

##Quality scores in FASTQ files

The fourth line of a FASTQ sequence format file encodes the quality score that tells us how good the sequence at a given position is (ie. how likely is it to be correct based on the instrument)

```{r}
library(seqinr)
library(gtools)
phred <- asc( s2c("DDDDCDEDCDDDDBBDDDCC@") ) - 33
phred
```

```{r}
prob <- 10**(-phred/10)
prob
```

##Population Scale Analysis

Read RNA-Seq count data with genotype information results table
```{r}
expr <- read.table("rs8067378_ENSG00000172057.6.txt")
head(expr)
```

```{r}
summary(expr)
```

```{r}
inds <- expr$geno == "G/G"
summary(expr[inds,"exp"])
```

```{r}
inds <- expr$geno == "A/G"
summary(expr[inds,"exp"])
```

```{r}
inds <- expr$geno == "A/A"
summary(expr[inds,"exp"])
```

```{r}
boxplot(exp ~ geno , data=expr)
```

```{r}
p <- boxplot(exp~geno, data=expr, xlab="rs8067378 genotype", ylab="ENSG00000172057.4 (RPKM)", notch=T)
```

```{r}
p
```


